### DISCUSSION TITLE
Support for node 20

### DISCUSSION THREAD

**Discussion by ventsyv** [supabase/supabase#16831] [Contribute to Supabase]
Trying to install fails:
```
npm install
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: supabase@0.0.0
npm ERR! notsup Not compatible with your version of node/npm: supabase@0.0.0
npm ERR! notsup Required: {"npm":">=9.0.0 <10.0.0","node":">=18.0.0 <19.0.0"}
npm ERR! notsup Actual:   {"npm":"9.8.0","node":"v20.5.1"}
```

Node 20 has been out for about 5 months now. It appears that updating the package.yaml file is all that's needed.
I built and ran the npm test targets and everything is passing.

**Comment by MadebyhumansAI** [comment]
Same here, strange thing is that it worked a few weeks ago but now after a redeploy it stopped working, will try with a few different docker images to see if it changes anything. @ventsyv How did you solve this?

2024-03-19T16:46:37.066Z [INFO]: # Executing command: npm clean-install
2024-03-19T16:46:41.069Z [WARNING]: npm
2024-03-19T16:46:41.075Z [WARNING]: ERR! code EBADENGINE
                                    npm ERR! engine Unsupported engine
                                    npm ERR! engine Not compatible with your version of node/npm: supabase@0.0.0
                                    npm ERR! notsup Not compatible with your version of node/npm: supabase@0.0.0
                                    npm ERR! notsup Required: {"npm":">=9.0.0 <10.0.0","node":">=18.0.0 <19.0.0"}
                                    npm ERR! notsup Actual:   {"npm":"10.2.4","node":"v18.19.1"}

**Reply by MadebyhumansAI** [reply]
I tried several different docker images until I found one where the npm and node versions meet the ones from the error message. The build is now successful again. I don't know exactly what caused it. Hopefully this will help someone in the future. This is the image that worked:
public.ecr.aws/sam/build-nodejs18.x:1.88.0-x86_64

**Reply by jasonradford115-ui** [reply]
Thanks
